src/pages/landing/components/RegisterModal.test.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+import { renderWithProviders } from 'utils/test-utils';
+import RegisterModal from './RegisterModal';
+
+
+describe('Landing page > Register modal', () => {
+
+  test('should not render register modal if show property is false', () => {
+
+    const props = {
+      show: false,
+      success: false,
+      transactionId: null,
+      onClose: () => {}
+    };
+
+    renderWithProviders(<RegisterModal {...props} />);
+
+    const dialogElement = screen.queryByRole('dialog');
```

Comment:
Isn't it too vague, referring using 'dialog'. Can you edit the component to incorporate testid and use it.